makeflow/src/dag_file.c

```diff
@@ -84,4 +87,32 @@ int dag_file_in_trans( const struct dag_file *f )
 		return 0;
 }
 
+int dag_file_mount_clean(struct dag *d) {
+	struct list *list;
+	struct dag_file *df;
+	if(!d) return 0;
+
+	list = dag_input_files(d);
+	if(!list) return 0;
+
+	list_first_item(list);
+	while((df = (struct dag_file *)list_next_item(list))) {
+		if(df->source) {
+			free(df->source);
+			df->source = NULL;
+		}
+		if(df->cache_name) {
+			free(df->cache_name);
+			df->cache_name = NULL;
+		}
+	}
+	list_delete(list);
+
+	if(d->cache_dir) {
```

Comment:
From @batrick  I've recently learned that you can do a free(ptr) on an NULL ptr. So these ifs can be removed.